File: 20-valid-parentheses/valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> s1;
        for(auto it:s){
            if(it=='('||it=='['||it=='{'){
                s1.push(it);
            }
            else{
                if(s1.empty()==true) return false;
                else if(it==')' && s1.top()!='(') return false;
                else if(it=='}' && s1.top()!='{') return false;
                else if(it==']' && s1.top()!='[') return false;
                else{s1.pop();}
            }
        }
        return s1.empty();
    }
};

File: 20-valid-parentheses/valid-parentheses.py
class Solution(object):
    def isValid(self, s) :
        st=[]
        for i in s:
            if i in"([{":
                st.append(i);
            elif st and ((st[-1]=='(' and i==')') or (st[-1]=='[' and i==']') or (st[-1]=='{' and i=='}')) :
                st.pop();
            else: 
                return False
        return not st
       
        